src/workflows/SnapshotActionMenu.tsx

```diff
@@ -23,23 +27,34 @@ export interface SnapshotActionMenuProps {
  * A kebab (vertical three-dot) menu that displays buttons to perform actions on
  * a workflow snapshot.
  *
- * Currently supported actions: delete
+ * Currently supported actions: edit permissions, delete snapshot
  */
 const SnapshotActionMenu = (props: SnapshotActionMenuProps): ReactNode => {
-  const { disabled, isSnapshotOwner, onDelete } = props;
+  const { disabled, isSnapshotOwner, onEditPermissions, onDelete } = props;
 
   const notSnapshotOwnerTooltip = 'You must be an owner of this snapshot';
 
   const menuContent = (
-    <MenuButton
-      disabled={!isSnapshotOwner}
-      tooltip={!isSnapshotOwner && notSnapshotOwnerTooltip}
-      tooltipSide='left'
-      onClick={onDelete}
-    >
-      {makeMenuIcon('trash')}
-      Delete snapshot
-    </MenuButton>
+    <>
+      <MenuButton
+        disabled={!isSnapshotOwner}
+        tooltip={!isSnapshotOwner && notSnapshotOwnerTooltip}
+        tooltipSide='left'
+        onClick={onEditPermissions}
+      >
+        {makeMenuIcon('cog')}
+        Edit permissions
```

Comment:
Similar to delete option, can this be renamed to `Edit snapshot permissions` to make it more clear that this only updates permissions for selected snapshot?